run.jl

```diff
@@ -1,66 +1,59 @@
 using ObjectDetector
 
-using Functors
-
-
 yolomod = YOLO.v3_608_COCO(batch=10, silent=true, use_gpu=false);
 
 yolomod = YOLO.v3_608_COCO(batch=1, silent=true, use_gpu=false);
 
 batch = emptybatch(yolomod);
 
+@showtime ret=yolomod(batch, detectThresh=0.5, overlapThresh=0.2);
+
 @showtime yolomod(batch, detectThresh=0.5, overlapThresh=0.8);
 
 @showtime yolomod(batch, detectThresh=0.5, overlapThresh=0.8);
+println()
 
-@showtime yolomod(batch, detectThresh=0.5, overlapThresh=0.8, show_timing=true);
+using AllocArrays, Adapt
 
-using AllocArrays
+# TODO- move to AllocArrays?
+"""
+    wrap_model(model; n_bytes=2^33, T=AllocArray)
 
-b = BumperAllocator(2^33); # 8.5 GB
+Wraps a model to use a bump allocator for temporary arrays. The `n_bytes` of memory will be preallocated upfront once when `wrap_model` is called, and re-used for every batch inferred over. The type `T` can be set to `CheckedAllocArray` for testing purposes.
 
-# T = CheckedAllocArray # for testing
-T = AllocArray
-batch_aa = T(batch);
-
-yolomod_b = deepcopy(yolomod)
-function aaize(obj)
-    fmap(x -> begin
-        x isa AbstractArray || return x
-        isbitstype(eltype(x)) || return x
-        return T(x)
-    end, obj; exclude=x -> x isa AbstractArray{<:Number} || x isa Function)
-end
-
-yolomod_b.chain = aaize(yolomod_b.chain)
-yolomod_b.W = Dict(k => T(v) for (k,v) in yolomod_b.W)
+## Example
 
-
-function bumper_yolomod(b, args...; kwargs...)
-    with_allocator(b) do
-        _ret = yolomod_b(args...; kwargs...)
-        ret = Array(_ret)
-        reset!(b)
-        return ret
+```julia
+yolomod = YOLO.v3_608_COCO(batch=1, silent=true, use_gpu=false);
+batch = emptybatch(yolomod);
+@time yolomod(batch) #  0.900718 seconds (6.89 k allocations: 6.295 GiB, 5.53% gc time)
+
+# Now let's use our bump allocator
+yolomod_aa = wrap_model(yolomod; T=AllocArray)
+@time yolomod_aa(batch) #  0.857606 seconds (8.74 k allocations: 697.273 KiB)
+
+```
+"""
+function wrap_model(model; n_bytes=2^33, T=AllocArray)
```

Comment:
I think we can pretty generically wrap Adapt-compatible models with a bump allocator that gets reused for every batch, which is kinda cool